### DISCUSSION TITLE
PostgreSQL package for nanos

### DISCUSSION THREAD

**Discussion by rrs45** [nanovms/ops#1139] [General]
Given that postgres is multi-process system, is it possible to run it on Nanos? There is a way to run it in single-user mode, will this allow it run on Nanos?

**Comment by francescolavra** [comment]
Unfortunately not, PostgreSQL is heavily dependent on `fork()`ing child processes to process DB requests, and fork() is not allowed in Nanos. It's also dependent on other things that are used in multi-process applications, such as shared memory segments and SIGCHLD signals, so it cannot be easily patched to make it runnable with a unikernel.
